### PR TITLE
Add helper methods to CsvOutput to bridge writing CSV Files and Iterators

### PR DESCRIPTION
**Pull request by ashleyheath**
Sometimes you already have an instance of a CsvFile or CsvIterator that you want to write out somewhere. This PR adds some static helper methods to CsvOutput to help support that.

### CODE REVIEW

File: modules/collect/src/main/java/com/opengamma/strata/collect/io/CsvOutput.java [hunk 1]
```diff
@@ -222,6 +434,22 @@ public void writeLines(Iterable<? extends List<String>> lines, boolean alwaysQuo
...
+  /**
+   * Writes multiple {@code CsvRow}s to the underlying.
+   * <p>
+   * The boolean flag controls whether each entry is always quoted or only quoted when necessary.
+   *
+   * @param rows  the rows to write
+   * @param alwaysQuote  when true, each column will be quoted, when false, quoting is selective
+   * @throws UncheckedIOException if an IO exception occurs
+   */
+  public void writeRows(Iterable<? extends CsvRow> rows, boolean alwaysQuote) {
```
Comment (jodastephen) [inline modules/collect/src/main/java/com/opengamma/strata/collect/io/CsvOutput.java]: * Method should be next to `writeRow` (after `writeLine`)
* `CsvRow` is final, so `? extends` won't add much.

File: modules/collect/src/main/java/com/opengamma/strata/collect/io/CsvOutput.java [hunk 2]
```diff
@@ -174,6 +174,218 @@ public static CsvOutput safe(Appendable underlying, String newLine, String separ
...
+  /**
+   * Writes the provided {@code CsvFile} to the underlying, using the system default line separator and using a
+   * comma separator.
+   * <p>
+   * See the standard quoting rules in the class-level documentation.
+   *
+   * @param file  the file whose contents to output
+   * @param underlying  the destination to write to
+   */
+  public static void writeStandard(CsvFile file, Appendable underlying) {
```
Comment (jodastephen) [inline modules/collect/src/main/java/com/opengamma/strata/collect/io/CsvOutput.java]: I think instance methods would be the better choice, as 12 methods should reduce to 2.

File: modules/collect/src/main/java/com/opengamma/strata/collect/io/CsvOutput.java (hunk 1 above)
Comment (ashleyheath) [reply modules/collect/src/main/java/com/opengamma/strata/collect/io/CsvOutput.java]: Keep writeLine_s_ and writeRow_s_ next to each other?

File: modules/collect/src/main/java/com/opengamma/strata/collect/io/CsvOutput.java (hunk 2 above)
Comment (ashleyheath) [reply modules/collect/src/main/java/com/opengamma/strata/collect/io/CsvOutput.java]: Add an 'alwaysQuote' param to each (for consistency) and make that four?

Comment (ashleyheath) [reply modules/collect/src/main/java/com/opengamma/strata/collect/io/CsvOutput.java]: Actually, ignore that, safe/standard is already on the instance, isn't it.

File: modules/collect/src/main/java/com/opengamma/strata/collect/io/CsvOutput.java [hunk 3]
```diff
@@ -222,6 +222,22 @@ public void writeLines(Iterable<? extends List<String>> lines, boolean alwaysQuo
     }
   }
 
+  /**
```
Comment (jodastephen) [inline modules/collect/src/main/java/com/opengamma/strata/collect/io/CsvOutput.java]: Move method below lines methods. ie. group all the lines methods, then all the rows methods, then all the cell methods, then the file/iterator ones

Comment (ashleyheath) [reply modules/collect/src/main/java/com/opengamma/strata/collect/io/CsvOutput.java]: Done.

**Pull request merged**